framework/auth/core.py

```diff
@@ -667,6 +667,7 @@ def add_unclaimed_record(self, node, referrer, given_name, email=None):
             'email': clean_email,
         }
         self.unclaimed_records[project_id] = record
+        self.save()
```

Comment:
We generally try to avoid implicit saves, except when *related* objects need to be saved. In this case, I think it's best to save explicitly.